enterprise/cypher/physical-planning/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/SlotAllocation.scala

```diff
@@ -113,15 +113,12 @@ object SlotAllocation {
 
   private def allocate(lp: LogicalPlan, nullable: Boolean, argument: Option[PipelineInformation]): PipelineInformation =
     lp match {
-      case _: Argument =>
-        argument.getOrElse(throw new InternalException("Found argument without Apply"))
-
       case leaf: NodeLogicalLeafPlan =>
         val pipeline = argument.getOrElse(PipelineInformation.empty)
         pipeline.newLong(leaf.idName.name, nullable, CTNode)
         pipeline
 
-      case SingleRow() =>
+      case _:SingleRow =>
```

Comment:
Previous behaviour for Apply was to throw an exception here 'Found argument without Apply'. That has been changed. Is that correct? I guess that assertion was only to catch presumably impossible cases where Argument was used like SingleRow, and now Argument can be used like SingleRow, so this differentiation disappears.

Reply:
Yep.